v2/internal/reconcilers/arm/azure_generic_arm_reconciler_instance.go

```diff
@@ -559,18 +560,29 @@ func (r *azureDeploymentReconcilerInstance) getStatus(ctx context.Context, id st
 	}
 
 	// Get the resource
-	retryAfter, err := r.ARMConnection.Client().GetByID(ctx, id, apiVersion, armStatus)
-	if err != nil {
-		return nil, retryAfter, errors.Wrapf(err, "getting resource with ID: %q", id)
-	}
-
-	if r.Log.V(Debug).Enabled() {
-		statusBytes, marshalErr := json.Marshal(armStatus)
-		if marshalErr != nil {
-			return nil, zeroDuration, errors.Wrapf(marshalErr, "serializing ARM status to JSON for debugging")
+	if operations.IsSupported(genruntime.ResourceOperationGet, r.Obj.GetSupportedOperations()) {
+		var retryAfter time.Duration
+		retryAfter, err = r.ARMConnection.Client().GetByID(ctx, id, apiVersion, armStatus)
+		if err != nil {
+			return nil, retryAfter, errors.Wrapf(err, "getting resource with ID: %q", id)
 		}
 
-		r.Log.V(Debug).Info("Got ARM status", "status", string(statusBytes))
+		if r.Log.V(Debug).Enabled() {
+			statusBytes, marshalErr := json.Marshal(armStatus)
+			if marshalErr != nil {
+				return nil, zeroDuration, errors.Wrapf(marshalErr, "serializing ARM status to JSON for debugging")
+			}
+
+			r.Log.V(Debug).Info("Got ARM status", "status", string(statusBytes))
+		}
+	} else if operations.IsSupported(genruntime.ResourceOperationHead, r.Obj.GetSupportedOperations()) {
```

Comment:
Why not make `IsSupported()` a member function of Operation, and add an interface for `GetSupportedOperations()` so that you can write this:

```suggestion
	} else if genruntime.ResourceOperationHead.IsSupportedBy(r.Obj) {
```

which I think would be clearer and more intention revealing.

Reply:
Yeah this is definitely clearer. Done.